python/ray/train/tests/test_mosaic_trainer.py

```diff
@@ -0,0 +1,208 @@
+    }
+
+    trainer = MosaicTrainer(
+        trainer_init_per_worker=trainer_init_per_worker,
+        trainer_init_config=trainer_init_config,
+        scaling_config=scaling_config,
+    )
+
+    trainer.fit()
+
+    # TODO : check the training duration once reporting/checkpoint has been integrated
+
+
+def test_init_errors(ray_start_4_cpus):
+    """Tests errors that may be raised when constructing MosaicTrainer. The error may
+    be due to bad `trainer_init_per_worker` function or missing requirements in the
+    `trainer_init_config` argument.
+    """
+    # invalid trainer init function -- no argument
+    def bad_trainer_init_per_worker_1():
+        pass
+
+    trainer_init_config = {
+        "max_duration": "1ba",
+    }
+
+    with pytest.raises(ValueError):
+        _ = MosaicTrainer(
+            trainer_init_per_worker=bad_trainer_init_per_worker_1,
+            trainer_init_config=trainer_init_config,
+            scaling_config=scaling_config,
+        )
+
+    # invalid trainer init function -- more than one argument
+    def bad_trainer_init_per_worker_1(a, b):
+        pass
+
+    with pytest.raises(ValueError):
+        _ = MosaicTrainer(
+            trainer_init_per_worker=bad_trainer_init_per_worker_1,
+            trainer_init_config=trainer_init_config,
+            scaling_config=scaling_config,
+        )
+
+    # datasets is not supported
+    with pytest.raises(ValueError, match=r".*dataset shards.*`prepare_dataloader`.*"):
+        _ = MosaicTrainer(
+            trainer_init_per_worker=trainer_init_per_worker,
+            trainer_init_config=trainer_init_config,
+            scaling_config=scaling_config,
+            datasets={"train": [1]},
+        )
+
+
+def test_loggers(ray_start_4_cpus):
+    class DummyLogger(LoggerDestination):
+        def fit_start(self, state: State, logger: Logger) -> None:
+            raise ValueError("Composer Logger object exists.")
+
+    class DummyCallback(Callback):
+        def fit_start(self, state: State, logger: Logger) -> None:
+            raise ValueError("Composer Callback object exists.")
+
+    # DummyLogger should not throw an error since it should be removed before `fit` call
+    trainer_init_config = {
+        "max_duration": "1ep",
+        "loggers": DummyLogger(),
+        "algorithms": [LabelSmoothing()],
+    }
+
+    trainer = MosaicTrainer(
+        trainer_init_per_worker=trainer_init_per_worker,
+        trainer_init_config=trainer_init_config,
+        scaling_config=scaling_config,
+    )
+
+    trainer.fit()
+
+    # DummyCallback should throw an error since it should not have been removed.
+    trainer_init_config["callbacks"] = DummyCallback()
+    trainer = MosaicTrainer(
+        trainer_init_per_worker=trainer_init_per_worker,
+        trainer_init_config=trainer_init_config,
+        scaling_config=scaling_config,
+    )
+
+    with pytest.raises(ValueError) as e:
+        trainer.fit()
+        assert e == "Composer Callback object exists."
```

Comment:
`e` is an `Exception`, but you are comparing against a string so this assert will fail.

Reply:
`with pytest.raises(ValueError, match="Composer Callback object exists")`